apps/web/src/pages/integrations/components/ConnectIntegrationForm.tsx

```diff
@@ -109,26 +109,50 @@ export function ConnectIntegrationForm({
             here.
           </a>
         </InlineDiv>
-        {provider?.credentials.map((credential) => (
-          <InputWrapper key={credential.key}>
-            <Controller
-              name={credential.key}
-              control={control}
-              render={({ field }) => (
-                <PasswordInput
-                  label={credential.displayName}
-                  required
-                  placeholder={credential.displayName}
-                  description={credential.description ?? ''}
-                  data-test-id={credential.key}
-                  error={errors[credential.key]?.message}
-                  {...field}
-                  {...register(credential.key, { required: `Please enter a ${credential.displayName.toLowerCase()}` })}
-                />
-              )}
-            />
-          </InputWrapper>
-        ))}
+        {provider?.credentials.map((credential) =>
+          isNeededToHide(credential.key) ? (
+            <InputWrapper key={credential.key}>
+              <Controller
+                name={credential.key}
+                control={control}
+                render={({ field }) => (
```

Comment:
Also looks like this is similar to both inputs, so maybe worth only doing the switch on the inner PasswordInput or Input components instead of the whole wrapper and Controller.

Reply:
You are totally right!